### PR TITLE
export RuleBuilder

### PR DESCRIPTION
**Pull request by willpuckett**
export RuleBuilder interface for easier use with Deno

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export * from './config/to-type-sequence'
 export * from './config/condition'
 
 // Rules
-export { rule } from './config/rule'
+export { rule, RuleBuilder } from './config/rule'
```
Comment (evan-liu) [inline src/index.ts:45]:
```suggestion
export type { BuildContext } from './utils/build-context'
export type { RuleBuilder } from './config/rule'
export { rule } from './config/rule'
```

Is `export type` enough? `BuildContext` may need be exported too.

Comment (willpuckett) [reply src/index.ts:45]: I think just RuleBuilder will do it... I'm using it with esm.sh so I'll confirm after the next release. Deno didn't like the require in `writeToProfile` so I'm using the below custom version. TypeScript is only complaining about `RuleBuilder` and I'm typing with a Karabiner config that `deno run -A index.ts` spit out now with no problems.

```ts
import {
  complexModifications,
  KarabinerConfig,
  ModificationParameters,
  Rule,
  RuleBuilder,
} from "https://esm.sh/karabiner.ts";

import { join } from "https://deno.land/std@0.192.0/path/mod.ts";

export async function writeToProfile(
  name: "--dry-run" | string,
  rules: Array<Rule | RuleBuilder>,
  parameters: ModificationParameters = {},
) {
  const karabinerConfigFile = join(
    Deno.env.get("HOME")!,
    ".config",
    "karabiner",
    "karabiner.json",
  );

  const config: KarabinerConfig = name === "--dry-run"
    ? { profiles: [{ name, complex_modifications: { rules: [] } }] }
    : (await import(karabinerConfigFile, { assert: { type: "json" } })).default;

  const profile = config?.profiles.find((v) => v.name === name);

  try {
    profile!.complex_modifications = complexModifications(rules, parameters);
  } catch (e) {
    console.error(e);
    return;
  }

  const json = JSON.stringify(config, null, 2);

  if (name === "--dry-run") {
    console.info(json);
    return;
  }

  Deno.writeTextFile(karabinerConfigFile, json);

  console.log(`✓ Profile ${name} updated.`);
}
```

**Comment by evan-liu** [comment]
Thanks for you contribution @willpuckett . I've made it a patch release [1.18.3](https://github.com/evan-liu/karabiner.ts/releases/tag/1.18.3), will work on a proper Deno support in a future release.

**Comment by evan-liu** [comment]
Hi @willpuckett , I've added Deno support, it can be now imported from `https://deno.land/x/karabinerts@1.19.0/deno.ts` if you'd like to give it a try. Thanks.

**Comment by willpuckett** [comment]
Wow that was fast. Just updated my config and it works great. Thanks so much for the attention.

I started exploring the mechanical keyboard community at the beginning of the year. As I started using some of my first builds, I tried some non-standard layouts and ended up settling on [Engram](https://engram.dev). More recently, I've been using home row mods on my mech keyboard. I tried emulating [this goku config](https://gist.github.com/eugenesvk/1f8fee49ce3fa5fe3cd4cce52085f7a5) for Engram, but being able to use the `withMapper` in karabiner.ts drastically simplifies things. 

As it stands, the home row mods are working well individually (using simlayer), but when I add additional conditional mappers to define combinations, things start freaking out (repo [here](https://github.com/willpuckett/karabiner-config-engram), offending lines commented out). If I watch the variables in Karabiner-EventViewer, some will remain set to 1 even after key releases which seems to be what is causing the unexpected behavior. 

I'll try and compare the generated karabiner.json from both methods and see if I can find the difference...

**Comment by evan-liu** [comment]
That is a very cool config. Glad to see the `withMapper` function being useful. 

For the simlayer variable issue, it could be the [default options](https://github.com/evan-liu/karabiner.ts/blob/main/src/config/simlayer.ts#L50), which can be overridden by the `options()` method. Please let me know if you found anything.

**Pull request merged**